File: gateway/src/main/java/com/example/gateway/controller/ItemController.java
package com.example.gateway.controller;

import com.example.api.ItemApi;
import com.example.api.dto.ItemDto;
import com.example.api.dto.ItemDtoWithBookings;
import com.example.gateway.client.ItemClient;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping(path = ItemApi.PATH)
@RequiredArgsConstructor
public class ItemController implements ItemApi {

    private final ItemClient itemClient;

    @Override
    public ItemDto create(ItemDto itemDto, Long userId) {
        return itemClient.create(itemDto, userId);
    }

    @Override
    public ItemDto update(Long id, ItemDto itemDto, Long userId) {
        return itemClient.update(id, itemDto, userId);
    }

    @Override
    public ItemDtoWithBookings findById(Long id, Long userId) {
        return itemClient.findById(id, userId);
    }

    @Override
    public List<ItemDtoWithBookings> findAllByOwnerId(Long userId, Integer from, Integer size) {
        return itemClient.findAllByOwnerId(userId, from, size);
    }

    @Override
    public List<ItemDto> search(String text, Integer from, Integer size) {
        return itemClient.search(text, from, size);
    }
}

File: gateway/src/main/kotlin/com/example/gateway/controller/ItemController.kt
package com.example.gateway.controller

import com.example.api.ItemApi
import com.example.api.dto.ItemDto
import com.example.api.dto.ItemDtoWithBookings
import com.example.gateway.client.ItemClient
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = [ItemApi.PATH])
class ItemController(
    val itemClient: ItemClient
) : ItemApi {

    override fun create(itemDto: ItemDto?, userId: Long?): ItemDto? {
        return itemClient.create(itemDto, userId)
    }

    override fun update(id: Long?, itemDto: ItemDto?, userId: Long?): ItemDto? {
        return itemClient.update(id, itemDto, userId)
    }

    override fun findById(id: Long?, userId: Long?): ItemDtoWithBookings? {
        return itemClient.findById(id, userId)
    }

    override fun findAllByOwnerId(userId: Long?, from: Int?, size: Int?): List<ItemDtoWithBookings?>? {
        return itemClient.findAllByOwnerId(userId, from, size)
    }

    override fun search(text: String?, from: Int?, size: Int?): List<ItemDto?>? {
        return itemClient.search(text, from, size)
    }
}
